test/blackbox-tests/test-cases/exec-watch/exec-watch-basic.t/run.t

```diff
@@ -10,16 +10,16 @@ between each change to its code.
   > EOF
 
   $ dune exec --watch ./foo.exe &
-  Success, waiting for filesystem changes...
+   changes...
```

Comment:
Why does it remove the entire sentence, minus the last word, here?

Reply:
No idea! It only happens in cram tests - not when running dune from a terminal. I've tried various `Console` functions to see if I could fix the test behaviour while also maintaining the behaviour when you run the command interactively but nothing works. Maybe something to do with how cram tests capture the output of commands? @rgrinberg any ideas?

Reply:
No idea, sorry. But shouldn't you restore the terminal after the process finishes?

Reply:
It's non-trivial to restore the terminal after the process finishes because dune doesn't wait for the process to finish and continues to run in watch mode, printing to the terminal itself while the new process runs in the background. This means that running programs that change terminal settings (such as programs with tuis) from exec watch mode will have a bad ux, especially because after every change dune will kill the process and launch a new process, possibly leaving the terminal in a bad state. It's definitely something we could improve in the future but not necessary to solve this issue.